elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/MetaDataStore.java

```diff
@@ -35,22 +37,47 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 /**
  * MetaDataStore is a in-memory data store that manage data models for an {@link AggregationDataStore}.
  */
-public class MetaDataStore extends HashMapDataStore {
+public class MetaDataStore implements DataStore {
     private static final Package META_DATA_PACKAGE = Table.class.getPackage();
 
     private static final List<Class<? extends Annotation>> METADATA_STORE_ANNOTATIONS =
             Arrays.asList(FromTable.class, FromSubquery.class, Subselect.class, javax.persistence.Table.class);
 
+    public static final Function<String, HashMapDataStore> ERROR_OUT = new Function<String, HashMapDataStore>() {
+        @Override
+        public HashMapDataStore apply(String key) {
+            throw new IllegalStateException("API version " + key + " not found");
+        }
+    };
+
+    private static final Function<String, HashMapDataStore> SETUP_NEW = new Function<String, HashMapDataStore>() {
+        @Override
+        public HashMapDataStore apply(String key) {
+            HashMapDataStore hashMapDataStore = new HashMapDataStore(META_DATA_PACKAGE);
+            EntityDictionary dictionary = new EntityDictionary(new HashMap<>());
+            ClassScanner.getAllClasses(META_DATA_PACKAGE.getName()).forEach(dictionary::bindEntity);
+            hashMapDataStore.populateEntityDictionary(dictionary);
+            return hashMapDataStore;
+        }
+    };
+
     @Getter
     private final Set<Class<?>> modelsToBind;
 
     private Map<Class<?>, Table> tables = new HashMap<>();
 
+    @Getter
+    private EntityDictionary dictionary = new EntityDictionary(new HashMap<>());
```

Comment:
I don't think we need this.  This appears to be unused.

Reply:
Its used in QueryEngine, ColumnVisitor etc. Renamed it to metadataDictionary to make it clear.